src/root/build.tt

```diff
@@ -524,11 +524,29 @@ END;
           <div class="d-flex flex-column  align-items-end">
             [% IF runcommandlog.start_time != undef %]
               <div>Started at [% INCLUDE renderDateTime timestamp = runcommandlog.start_time; %]</div>
-              [% IF runcommandlog.end_time != undef %]
-                <div>Ran for [% INCLUDE renderDuration duration = runcommandlog.end_time - runcommandlog.start_time %]</div>
-              [% ELSE %]
-                <div>Running for [% INCLUDE renderDuration duration = curTime - runcommandlog.start_time %]</div>
-              [% END %]
+              <div class="d-flex flex-column align-items-end">
+                [% IF runcommandlog.end_time != undef %]
+                  Ran for [% INCLUDE renderDuration duration = runcommandlog.end_time - runcommandlog.start_time %]
+                    [% IF runcommandlog.uuid != undef %]
+                      [% runLog = c.uri_for('/build', build.id, 'runcommandlog', runcommandlog.uuid) %]
+                      <div>
+                        <a class="btn btn-secondary btn-sm" href="[% runLog %]">pretty</a>
+                        <a class="btn btn-secondary btn-sm" href="[% runLog %]/raw">raw</a>
+                        <a class="btn btn-secondary btn-sm" href="[% runLog %]/tail">tail</a>
+                      </div>
+                    [% END %]
+                [% ELSE %]
+                  Running for [% INCLUDE renderDuration duration = curTime - runcommandlog.start_time %]
+                    [% IF runcommandlog.uuid != undef %]
+                      [% runLog = c.uri_for('/build', build.id, 'runcommandlog', runcommandlog.uuid) %]
+                      <div>
+                        <a class="btn btn-secondary btn-sm" href="[% runLog %]">pretty</a>
+                        <a class="btn btn-secondary btn-sm" href="[% runLog %]/raw">raw</a>
+                        <a class="btn btn-secondary btn-sm" href="[% runLog %]/tail">tail</a>
+                      </div>
+                    [% END %]
```

Comment:
It looks like this code block can be moved outside this `IF / ELSE / END` and then not duplicated in both branches. I might be missing something, though.